File: Android/app/src/main/java/com/pokeapi/lpiem/pokeapiandroid/Model/SocialNetworks/GoogleProfile.java
package com.pokeapi.lpiem.pokeapiandroid.Model.SocialNetworks;

import java.net.URI;

public class GoogleProfile {
    //Pseudo
    private String username;
    //E-mail
    private String email;
    //Prénom
    private String firstName;
    //Nom de famille
    private String familyName;
    //Surnom
    private String givenName;
    //Identifiant
    private String id;
    //URI du compte
    private URI accountURI;

    public GoogleProfile(String username, String email, String firstName, String familyName, String givenName, String id, URI accountURI) {
        this.username = username;
        this.email = email;
        this.firstName = firstName;
        this.familyName = familyName;
        this.givenName = givenName;
        this.id = id;
        this.accountURI = accountURI;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getFamilyName() {
        return familyName;
    }

    public void setFamilyName(String familyName) {
        this.familyName = familyName;
    }

    public String getGivenName() {
        return givenName;
    }

    public void setGivenName(String givenName) {
        this.givenName = givenName;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public URI getAccountURI() {
        return accountURI;
    }

    public void setAccountURI(URI accountURI) {
        this.accountURI = accountURI;
    }
}

File: Android/app/src/main/java/com/pokeapi/lpiem/pokeapiandroid/Model/SocialNetworks/GoogleProfile.kt
package com.pokeapi.lpiem.pokeapiandroid.Model.SocialNetworks

import android.net.Uri

class GoogleProfile(//Pseudo
        _email: String?, //Email
        _firstName: String?, //Prénom
        _familyName: String?, //Nom de famille
        _givenName: String?, //Identifiant
        _id: String?, //URI du compte
        _accountURI: Uri?
) {
    private var email:String = _email!!
    private var firstName:String = _firstName!!
    private var familyName:String = _familyName!!
    private var givenName:String = _givenName!!
    private var id:String = _id!!
    private var accountURI:Uri = _accountURI!!

    //Getters/Setters
    var Email: String
        get() = this.email
        set(newValue){
            this.email = newValue
        }
    var FirstName: String
        get() = this.firstName
        set(newValue){
            this.firstName = newValue
        }
    var FamilyName: String
        get() = this.familyName
        set(newValue){
            this.familyName = newValue
        }
    var GivenName: String
        get() = this.givenName
        set(newValue){
            this.givenName = newValue
        }
    var Id: String
        get() = this.id
        set(newValue){
            this.id = newValue
        }
    var AccountURI: Uri
        get() = this.accountURI
        set(newValue){
            this.accountURI = newValue
        }

}
